File: frontend/src/lib/project-state.ts
import { components } from "@/generated";

export enum ProjectState {
    Unknown = "Unknown",
    Running = "Running",
    Starting = "Starting",
    Stopped = "Stopped",
    Deleting = "Deleting"
}

export function getProjectStateFromNumber(stateNum?: components["schemas"]["ProjectState"]): ProjectState {
    switch (stateNum) {
        case "Unknown": return ProjectState.Unknown;
        case "Running": return ProjectState.Running;
        case "Starting": return ProjectState.Starting;
        case "Stopped": return ProjectState.Stopped;
        case "Deleting": return ProjectState.Deleting;
        default: return ProjectState.Unknown;
    }
}

File: backend/Api.Data/ProjectState.cs
using System.Text.Json.Serialization;

namespace Api.Data;

[JsonConverter(typeof(JsonStringEnumConverter<ProjectState>))]
public enum ProjectState
{
    Unknown = 0,
    Running = 1,
    Starting = 2,
    Stopped = 3,
    Deleting = 4,
}